  Since built-in "history" - and command-line editor capabilities - are
  much too simplistic, it is better to rely on "Rlwrap" utility instead
  when working under control of Linux OS.
diff -Nur pforth_v24/csrc/pf_io.c pforth_v24.new/csrc/pf_io.c
--- pforth_v24/csrc/pf_io.c	2009-02-20 19:38:25.000000000 +0100
+++ pforth_v24.new/csrc/pf_io.c	2010-06-19 21:48:33.128230901 +0200
@@ -90,6 +90,7 @@
 #define SPACE      (0x20)
 #define BACKSPACE  (0x08)
 #define DELETE     (0x7F)
+#define LF         (0x0A)
 cell ioAccept( char *buffer, cell maxChars )
 {
 	int c;
@@ -136,7 +137,7 @@
 
 gotline:
 	sdDisableInput();
-	sdTerminalEcho( SPACE );
+	sdTerminalEcho( LF );
 
 /* NUL terminate line to simplify printing when debugging. */
 	if( len < maxChars ) p[len] = '\0';
diff -Nur pforth_v24/fth/loadp4th.fth pforth_v24.new/fth/loadp4th.fth
--- pforth_v24/fth/loadp4th.fth	2008-07-21 04:58:11.000000000 +0200
+++ pforth_v24.new/fth/loadp4th.fth	2010-06-19 21:42:46.948266721 +0200
@@ -40,7 +40,7 @@
 include? words.like wordslik.fth
 include? trace   trace.fth
 include? ESC[    termio.fth
-include? HISTORY history.fth
+\ include? HISTORY history.fth
 include? SDAD    savedicd.fth
 
 map
